spyder/app/utils.py

```diff
@@ -325,3 +334,68 @@ def create_window(WindowClass, app, splash, options, args):
     if not running_under_pytest():
         app.exec_()
     return main
+
+
+class SpyderInstallerError(Exception):
+    """
+    Base class for installer error; do not use directly.
+    Exit Spyder with code 1.
+    """
+    def __init__(self, *args, **kwargs):
+        if not running_installer_test():
+            # Don't do anything
+            return
+
+        msg, args = self._msg(*args)
+
+        super().__init__(msg, *args, **kwargs)
+
+        # Print the stack trace and provided message,
+        # then exit Spyder with code 1.
+        # Note: raising exceptions will not print the stack trace
+        stack = ['Traceback (most recent call last):\n']
+        stack.extend(traceback.format_stack()[:-2])
+        stack.extend(traceback.format_exception_only(type(self), self))
+        print(''.join(stack))
```

Comment:
```suggestion
        print(''.join(stack))  # spyder: test-skip
```

This is why our fast tests are failing.

Reply:
Actually, I suppose the logger should be preferred over print.